tracing/src/main/java/com/palantir/tracing/TraceState.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.nio.ByteBuffer;
+import java.util.Random;
+import java.util.UUID;
+import org.immutables.value.Value;
+
+/**
+ * A container object used to hold information about call tracing.
+ */
+@JsonDeserialize(as = ImmutableTraceState.class)
+@JsonSerialize(as = ImmutableTraceState.class)
+@Value.Immutable
+@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE)
+public abstract class TraceState {
+
+    private static final Random RANDOM = new Random();
+
+    /**
+     * Return a description of the operation for this event.
+     */
+    public abstract String getOperation();
+
+    /**
+     * Return a globally unique identifier representing this call trace.
+     */
+    @SuppressWarnings("checkstyle:designforextension")
+    @Value.Default
+    public String getTraceId() {
+        return randomId();
+    }
+
+    public static final Builder builder() {
+        return new Builder();
+    }
+
+    private static String randomId() {
```

Comment:
In particular, https://github.com/rholder/fauxflake/blob/master/fauxflake-core/src/main/java/com/github/rholder/fauxflake/provider/boundary/FlakeEncodingProvider.java